portset/portset.go

```diff
@@ -0,0 +1,239 @@
+package portset
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os/user"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/aporeto-inc/trireme-lib/cache"
+	"github.com/bvandewalle/go-ipset/ipset"
+	"go.uber.org/zap"
+)
+
+const (
+	procNetTCPFile                    = "/proc/net/tcp"
+	portSetUpdateIntervalMilliseconds = 1000
+	portEntryTimeout                  = portSetUpdateIntervalMilliseconds * 3
+	uidFieldOffset                    = 7
+	procHeaderLineNum                 = 0
+	portOffset                        = 1
+	ipPortOffset                      = 1
+	sockStateOffset                   = 3
+	sockListeningState                = "0A"
+	hexFormat                         = 16
+	integerSize                       = 64
+	minimumFields                     = 2
+)
+
+// portSetInstance : This type contains look up tables
+// to help update the ipset portsets.
+type portSetInstance struct {
+	userPortSet *cache.Cache
+	userPortMap *cache.Cache
+}
+
+// deletes the port entry in the portset when the key uid:port
+// expires
+func expirer(c cache.DataStore, id interface{}, item interface{}) {
+
+	userPort := strings.Split(id.(string), ":")
+	portSetObject := item.(*portSetInstance)
```

Comment:
This needs to be checked if portSetObject is not nil, if nil continue otherwise panic